tests/units/anta_tests/test_bfd.py

```diff
@@ -0,0 +1,248 @@
+                                        "status": "up",
+                                        "remoteDisc": 0,
+                                        "lastDown": 1703657258.652725,
+                                        "l3intf": "Ethernet2",
+                                    }
+                                }
+                            },
+                        },
+                        "ipv6Neighbors": {},
+                    }
+                }
+            },
+            {
+                "utcTime": 1703658481.8778424,
+            },
+        ],
+        "inputs": {},
+        "expected": {
+            "result": "failure",
+            "messages": ["Following BFD peers were down:\n192.0.255.7 in default VRF has remote disc 0.\n192.0.255.71 in default VRF has remote disc 0."],
+        },
+    },
+    {
+        "name": "failure-last-down",
+        "test": VerifyBFDPeersHealth,
+        "eos_data": [
+            {
+                "vrfs": {
+                    "default": {
+                        "ipv4Neighbors": {
+                            "192.0.255.7": {
+                                "peerStats": {
+                                    "": {
+                                        "status": "up",
+                                        "remoteDisc": 3940685114,
+                                        "lastDown": 1703657258.652725,
+                                        "l3intf": "",
+                                    }
+                                }
+                            },
+                            "192.0.255.71": {
+                                "peerStats": {
+                                    "": {
+                                        "status": "up",
+                                        "remoteDisc": 3940685114,
+                                        "lastDown": 1703657258.652725,
+                                        "l3intf": "",
+                                    }
+                                }
+                            },
+                            "192.0.255.17": {
+                                "peerStats": {
+                                    "": {
+                                        "status": "up",
+                                        "remoteDisc": 3940685114,
+                                        "lastDown": 1703657258.652725,
+                                        "l3intf": "",
+                                    }
+                                }
+                            },
+                        },
+                        "ipv6Neighbors": {},
+                    }
+                }
+            },
+            {
+                "utcTime": 1703667348.111288,
+            },
+        ],
+        "inputs": {"down_threshold": 2},
+        "expected": {
+            "result": "failure",
+            "messages": [
+                "Following BFD peers were down:\n192.0.255.7 in default VRF was down 3 hours ago.\n"
```

Comment:
Why it says the peer was down 3 hours ago when the provided threshold is 2?

Reply:
Fixed... Earlier i was checking that if peer was down for more then threshold hours then failing. Now changed it to if peer is down within threshold hours then fail it. If threshold is 2 hours and and peer was down 1 hour ago then test will fail. If peer was down 3 hours ago then it wont fail.